lib/get-config.js

```diff
@@ -345,6 +361,50 @@ function getConfigForFile(config, filePath) {
   return config;
 }
 
+/**
+ * Sets the severity and config on the rule object.
+ * Eg:
+ * {'no-implicit-this': 'warn'} -> {'no-implicit-this': { severity: 'warn', config: true }}
+ * { 'no-implicit-this': [ 'warn', { allow: [ 'fooData' ] } ] }
+ * would become:
+ * { 'no-implicit-this': { severity: 'warn', config: 'allow': [ 'fooData' ] } }
+ * {
+ *  'some-rule': 'lol',                   // -> { severity: 2, config: 'lol' }
+ *  'other-thing': ['wat', 'is', 'this'], // { severity: 2, config: ['wat', 'is', 'this'] }
+ *  'hmm-thing-here': { zomg: 'lol' },    // -> { severity: 2, config: { zomg: 'lol' } }
+ *  'another-thing-there': 'off',    // -> { severity: 0, config: false }
+ * }
+ * @param {*} configData
+ */
+function processRules(config) {
+  for (let key in config.rules) {
+    let severityConfig;
+    let ruleData = config.rules[key];
+    // defaults to error always.
+    let ruleConfig = {
+      severity: ruleData === false ? IGNORE_SEVERITY : ERROR_SEVERITY,
+      config: ruleData,
+    };
+    // In case of {'no-implicit-this': 'off|warn|error'}
+    if (typeof ruleData === 'string') {
+      severityConfig = _determineConfigForSeverity(ruleData);
+      if (severityConfig) {
+        ruleConfig = severityConfig;
+      }
+    } else if (Array.isArray(ruleData)) {
+      // array of severity and custom rule config
+      let severity = ruleData[0];
+      severityConfig = _determineConfigForSeverity(severity);
+      if (severityConfig) {
+        ruleConfig.severity = severityConfig.severity;
+        ruleConfig.config = ruleData[1];
+      }
+    }
+    config.rules[key] = ruleConfig;
+  }
+  return Object.assign({}, config);
```

Comment:
We return a new object here, but we've already mutated it just above. We should do one thing, or the other, not both...